### ISSUE TITLE
Compiling for Ubuntu 16 / Mint - Uses all available memory and crashes system.

### ISSUE THREAD

**Issue by diversivolent** [Cockatrice/Cockatrice#2595]
<b>OS:</b>
Linux Mint 18 Cinnamon 64-bit
Cinnamon 3.0.7
Linux Kernel 4.4.0-72-generic

<b>Cockatrice version:</b>
Thopter pie network rev 2. dc6c375
Looks like it also happens with Thopter pie network rev 1 - ef7670a. 
Haven't checked earlier releases.
__________

Hello!

I'm trying to compile for Linux Mint / Ubuntu 16 as per the instructions here: 
https://github.com/Cockatrice/Cockatrice/wiki/Compiling-Cockatrice-(Linux)

It seems there is something wrong with the current build, as when I run make, it consumes 100% of available system memory (8GB on my laptop) CPU goes to 100% and hangs my system.

I've replicated this with a fresh install of Linux Mint 18 in a VM, same thing happens, CPU goes to 100%, all virtual memory is utilized, and the system crashes. 

Basically I issue the "make package -j" command, it proceeds until around ~60%, and then my entire system locks up and is unrecoverable.

Any ideas? I've had zero issues compiling on Mint 18 with previous earlier versions of Cockatrice.

**Comment by Daenyth** [comment]
Try running `make` before running `make package`, and don't use `-j`. Does that help? Off hand I don't think we greatly changed things...

**Comment by ZeldaZach** [comment]
Using `make package -j` will use _all_ available memory on the computer until it does hang. Make sure to set a value for the J, most often it's set to a value of 8 (`-j 8`)

**Comment by diversivolent** [comment]
Thanks... you can close this. 
Running make first and then making the packing with an argument for the -j switch worked fine.

Not sure why Mint 18 is totally locking up now, since in the past I've always followed the instructions on the installation page. Seems its clearly not related to Cockatrice since it compiles file if you don't run make with unlimited parallel jobs.

Perhaps maybe it would be good to update the compile instructions for Ubuntu 16 to give an argument for -j?

**Comment by Daenyth** [comment]
`-j` tells make to use N threads where N=cores. I suppose that it consumed all your system memory. Running without it just makes it single threaded

**Comment by Daenyth** [comment]
Manually passing `-j N` you can tell it how many. There's really not a set number; on my machine using all cores is fine. I suppose we can change it to omit `-j`, just that people who don't know better will have a much slower compile
